### DISCUSSION TITLE
Allow for custom OAuth scopes when self-hosing

### DISCUSSION THREAD

**Discussion by 0x77dev** [langfuse/langfuse#2376] [Ideas]
### Describe the feature or potential improvement

Extend this to include custom OAuth scopes using env variable: https://github.com/langfuse/langfuse/blob/main/packages/shared/src/server/auth.ts#L68C40-L68C60

For example, Clerk does not allow the `openid` scope, and `profile` is sufficient for auth.js.

### Additional information

- https://clerk.com/docs/advanced-usage/clerk-idp#use-clerk-as-an-o-auth-2-provider
- https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider
- https://langfuse.com/docs/deployment/self-host#sso

**Comment by marcklingen** [comment]
Open for PR on this. I think using `env.AUTH_CUSTOM_SCOPES ?? "openid email profile"` makes sense.

**Reply by marcklingen** [reply]
@0x77dev I implemented this here, will be part of the next release: https://github.com/langfuse/langfuse/pull/2459

**Reply by 0x77dev** [reply]
Hey @marcklingen, thank you so much for this one, there is still a need to configure additional options. For example, the `idToken` is not supported by Clerk and is currently hardcoded: https://github.com/langfuse/langfuse/blob/main/packages/shared/src/server/auth/customSsoProvider.ts#L20C1-L20C19

It would be beneficial to have the ability to modify anything in the OAuth provider, as described in the Next.js Auth documentation: https://next-auth.js.org/configuration/providers/oauth

If you'd like, I can create a pull request to address this.

**Reply by marcklingen** [reply]
sounds good, please tag me on a PR on this

you plan to add these as additional env vars?
